### PR TITLE
[java] Fix #4487: A false-positive about UnnecessaryConstructor and @Inject and @Autowired

### LINKED ISSUE #4487: [java] UnnecessaryConstructor: false-positive with @Inject and @Autowired
**Affects PMD Version:7.0.0-rc1** 

**Rule: UnnecessaryConstructor**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryconstructor>

**Description:**
The first code sample uses the annotation `com.google.inject.Inject`, and the second one uses `javax.inject.Inject`. The analysis results of the two cases should be equivalent because these two annotations are equivalent. However, PMD reports a warning in the first sample and no warnings in the second one. Actually, PMD should not report a warning in `Case 1` like the `Case 2` because it has a injection here.

The exact same issue applies to Spring's `@Autowired`.

**Code Sample demonstrating the issue:**

**Case 1**
```java
import com.google.inject.Inject;
public class Foo {
    private Foo() {}
}

public class Bar extends Foo {
    @Inject
    public Bar() {} // report a warnning
}
```

**Case 2**
```java
import javax.inject.Inject;
public class Foo {
    private Foo() {}
}

public class Bar extends Foo {
    @Inject
    public Bar() {}  //  report no warnning
}

```

**Expected outcome:** PMD reports a violation at line 8 in the first sample, but that's wrong. That's a false positive. PMD should report no violation if PMD sees `com.google.inject.Inject`.

**Running PMD through:** *CLI*

### PR DESCRIPTION
**Pull request by LynnBroe**
## Describe the PR

This PR excludes `com.google.inject.Inject` and `org.springframework.beans.factory.annotation.Autowired`  which cause FP.

## Related issues

- Fixes #4487 

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 13 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/835bdd1414534225a7448ae5b50ae8b5/diff1/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryConstructorRule.java [hunk 1]
```diff
@@ -35,7 +35,8 @@ public class UnnecessaryConstructorRule extends AbstractIgnoredAnnotationRule {
 
     @Override
     protected Collection<String> defaultSuppressionAnnotations() {
-        return Collections.singletonList("javax.inject.Inject");
+        return Arrays.asList("javax.inject.Inject",
+                "com.google.inject.Inject");
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryConstructorRule.java:39]: we should probably also include Spring's `@Autowired` here…

Comment (LynnBroe) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryConstructorRule.java:39]: Thanks for comment. I will add Spring's @Autowired later

**Comment by jsotuyod** [comment]
Thanks!

**Pull request merged**